libbeat/esleg/eslegclient/connection.go

```diff
@@ -75,6 +76,8 @@ type ConnectionSettings struct {
 
 	Timeout         time.Duration
 	IdleConnTimeout time.Duration
+
+	encodedAPIKey string // Base64-encoded API key
```

Comment:
Maybe we want to move the field to `Connection` ?

Reply:
Sure, I guess because you want to keep `ConnectionSettings` just as the public settings API?

Reply:
Done in dc4e921.

Reply:
> Sure, I guess because you want to keep ConnectionSettings just as the public settings API?

yep.